assets/javascripts/discourse/services/chat-message-visibility-observer.js

```diff
@@ -13,6 +15,11 @@ export default class ChatMessageVisibilityObserver extends Service {
   _observerCallback(entries) {
     entries.forEach((entry) => {
       entry.target.dataset.visible = entry.isIntersecting;
+
+      if (entry.isIntersecting && !isTesting()) {
+        const chatService = getOwner(this).lookup("service:chat");
```

Comment:
I wonder if it would be cleaner to inject this into the ChatMessageVisibilityObserver?

```javascript
import { inject as service } from "@ember/service";

export default class ChatMessageVisibilityObserver extends Service {
  @service("chat") chatService;
  ...
```

Reply:
This didn't work for me, but now I realize I also need to decorate the callback function with a `@bind`.